### PR TITLE
Flex spec v2

### PR DESCRIPTION
**Pull request by br648**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

Update to include the lastest Flex spec v2.

### CODE REVIEW

**Review by JymDyerIBI — APPROVED** [review]
These look good.

File: .github/workflows/build-docker.yml
Comment (binh-dam-ibigroup) [inline .github/workflows/build-docker.yml]: Is this change needed after approval?

File: src/main/resources/gtfs/gtfs.yml
Comment (binh-dam-ibigroup) [inline src/main/resources/gtfs/gtfs.yml]: "Multiple routes may use the same *location*". And also, please clarify "The stop_id is referenced from the stops.txt file." (This field is about `location_id`)

File: pom.xml
Comment (binh-dam-ibigroup) [inline pom.xml]: (Remember to update GTFS-lib version number.)

**Review by binh-dam-ibigroup — APPROVED** [review]
Approving subject to the comments indicated.

File: .github/workflows/build-docker.yml
Comment (br648) [reply .github/workflows/build-docker.yml]: @miles-grant-ibigroup FYI I'm going to remove this. Once this PR is merged you can use dev-flex.

File: src/main/resources/gtfs/gtfs.yml
Comment (br648) [reply src/main/resources/gtfs/gtfs.yml]: @binh-dam-ibigroup On review I should probably revert this back and create a new gtfsFlex.yml file. However, this current file is only referenced once in DT and the new gtfsFlex.yml wouldn't be used. So, I'm just going to revert this file back and not create the new file. Thoughts?

Comment (binh-dam-ibigroup) [reply src/main/resources/gtfs/gtfs.yml]: Sure

File: pom.xml
Comment (br648) [reply pom.xml]: Fixed in: https://github.com/ibi-group/datatools-server/pull/599/commits/10d924c72b5e18346ba2146ddb90601d35eda486

File: src/main/resources/gtfs/gtfs.yml
Comment (br648) [reply src/main/resources/gtfs/gtfs.yml]: Fixed in: https://github.com/ibi-group/datatools-server/pull/599/commits/10d924c72b5e18346ba2146ddb90601d35eda486

File: .github/workflows/build-docker.yml
Comment (br648) [reply .github/workflows/build-docker.yml]: Fixed in: https://github.com/ibi-group/datatools-server/pull/599/commits/10d924c72b5e18346ba2146ddb90601d35eda486

**Review by JymDyerIBI — COMMENTED** [review]
Changes look good.

**Pull request merged**
